tests/ubench/mech_vec.cpp

```diff
@@ -12,6 +12,18 @@ using namespace arb;
 using backend = arb::multicore::backend;
 using fvm_cell = arb::fvm_lowered_cell_impl<backend>;
 
+int find_mechanism(const std::string &mech_name, fvm_cell &cell) {
```

Comment:
The error handling code is the same for each benchmark (i.e. print error message and `exit(1)`), so I would put it in this function, and also return the `mechanism_ptr` directly.
```C++
mechanism_ptr& find_mechanism(const std::string& name, fvm_cell& cell) {
    auto &mechs = cell.mechanims();
    auto it = std::find_if(mechs.begin(),
                           mechs.end(),
                           [&](mechanism_ptr& m){return m->internal_name()==name;});
    if (it==mechs.end()) {
        std::cerr << "couldn't find mechanism with name " << name << "\n";
        exit(1);
    }   
    return *it;
}

// which can be called directly
auto& m = find_mechanism("hh", cell);
```
I haven't tested this, but you get the idea!

Reply:
That makes a lot of sense.